src/http.cc

```diff
@@ -1369,7 +1369,7 @@ HttpStateData::writeReplyBody()
     const char *data = inBuf.rawContent();
     int len = inBuf.length();
     addVirginReplyBody(data, len);
-    inBuf.consume(len);
+    inBuf.chop(len);
```

Comment:
> I suspect that we need to make sure that a zero-length SBuf calls clear() before any updates

Finally no need to modify more the SBuf at all. The new SBuf::cow calls MemBlob::crop to clear blob and this is enough.

Commit be940427 put back SBuf::consume calls instead of SBuf::chop.

Reply:
> no need to modify more the SBuf at all. The new SBuf::cow calls MemBlob::crop to clear blob and this is enough.

I am skeptical that those cow() modifications would be enough to cover all use cases similar to the one you were trying to address by replacing consume() with chop(). However, I am resolving this particular change request. There is another still-open change request that deals with similar problem, and I will continue the discussion there.